packages/typespec-ts/src/modular/helpers/operationHelpers.ts

```diff
@@ -869,7 +898,9 @@ export function getResponseMapping(
         (property.restApiName === "message" ||
           property.restApiName === "messages") &&
         (property.type.name === "ChatMessage" ||
-          property.type.elementType?.name === "ChatMessage")
+          property.type.elementType?.name === "ChatMessage" ||
+          property.type.name === "ChatResponseMessage" ||
+          property.type.elementType?.name === "ChatResponseMessage")
```

Comment:
There is a circular reference in `ChatResponseMessage` and we haven't fix that issue yet, have to add the temporary work around here.